File: quick-fix.js
const fs = require('fs');
const path = require('path');

function quickFix(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let originalContent = content;
    
    // Fix common issues
    content = content
      // Fix HTML entities in imports
      .replace(/&quot;/g, '"')
      .replace(/&lt;/g, '<')
      .replace(/&gt;/g, '>')
      .replace(/&amp;/g, '&')
      
      // Fix missing semicolons after "use client"
      .replace(/"use client",\s*\n/g, '"use client";\n')
      
      // Fix interface property syntax
      .replace(/(\w+):\s*(\w+);,\s*\n/g, '$1: $2;\n')
      .replace(/(\w+):\s*(\w+);\s*\n/g, '$1: $2;\n')
      
      // Fix missing semicolons in interfaces
      .replace(/(\w+)\s*:\s*(\w+)\s*\n/g, '$1: $2;\n')
      
      // Fix duplicate semicolons
      .replace(/;;/g, ';')
      
      // Fix missing commas in arrays
      .replace(/(\w+)\s*\n\s*\]/g, '$1,\n  ]')
      
      // Fix missing closing tags
      .replace(/<div([^>]*)>\s*([^<]*)\s*$/gm, '<div$1>$2</div>')
      
      // Fix JSX attribute syntax
      .replace(/className\s*=\s*"([^"]*)\s*([^"]*)"/g, 'className="$1 $2"')
      
      // Fix missing return statements
      .replace(/const\s+(\w+)\s*=\s*\(\s*\)\s*=>\s*\{([^}]+)\}\s*$/gm, (match, funcName, body) => {
        if (!body.trim().includes('return')) {
          return `const ${funcName} = () => {\n  return ${body.trim()};\n}`;
        }
        return match;
      });
    
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    return false;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

function walkDirectory(dir) {
  const files = fs.readdirSync(dir);
  let fixedCount = 0;
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory() && !file.startsWith('.') && file !== 'node_modules') {
      fixedCount += walkDirectory(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      if (quickFix(filePath)) {
        fixedCount++;
      }
    }
  });
  
  return fixedCount;
}

console.log('Starting quick fixes...');
const fixedCount = walkDirectory('./app');
console.log(`Fixed ${fixedCount} files in app directory`);

const srcFixedCount = walkDirectory('./src');
console.log(`Fixed ${srcFixedCount} files in src directory`);

console.log('Quick fixes completed!');

File: quick_fix.py
#!/usr/bin/env python3

import os
import re
import glob

def quick_fix():
    """Quick fix for common syntax errors"""
    
    # Find all TypeScript/JavaScript files
    files = glob.glob("*.{ts,tsx,js,jsx}")
    files.extend(glob.glob("src/**/*.{ts,tsx,js,jsx}", recursive=True))
    files = [f for f in files if not f.startswith('node_modules') and not f.startswith('.next') and not f.startswith('.git')]
    
    print(f"Found {len(files)} files to process")
    
    for file_path in files:
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original_content = content
            
            # Fix double commas
            content = re.sub(r',\s*,', ',', content)
            content = re.sub(r',\s*\)', ')', content)
            content = re.sub(r'\(\s*,', '(', content)
            
            # Fix missing function declarations
            if content.strip().startswith('return (') and 'function' not in content:
                content = 'export default function Component() {\n' + content + '\n}'
            
            # Fix missing React import
            if 'import React' not in content and ('from "react"' in content or 'from \'react\'' in content):
                content = 'import React from "react";\n' + content
            
            # Fix trailing commas
            content = re.sub(r',\s*}', '}', content)
            content = re.sub(r',\s*]', ']', content)
            
            # Fix JSX syntax
            content = re.sub(r'<(\w+),\s*', r'<\1 ', content)
            content = re.sub(r',\s*>', '>', content)
            
            if content != original_content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Fixed {file_path}")
                
        except Exception as e:
            print(f"Error processing {file_path}: {e}")

if __name__ == "__main__":
    quick_fix()